src/oralb_ble/parser.py

```diff
@@ -287,3 +293,26 @@ def _start_update(self, service_info: BluetoothServiceInfo) -> None:
         self.update_binary_sensor(
             str(OralBBinarySensor.BRUSHING), bool(state == 3), None, "Brushing"
         )
+
+    async def async_poll(self, ble_device: BLEDevice) -> SensorUpdate:
+        """
+        Poll the device to retrieve any values we can't get from passive listening.
+        """
+        client = await establish_connection(BleakClient, ble_device, ble_device.address)
+        try:
+            battery_char = client.services.get_characteristic(CHARACTERISTIC_BATTERY)
+            battery_payload = await client.read_gatt_char(battery_char)
+            pressure_char = client.services.get_characteristic(CHARACTERISTIC_PRESSURE)
+            pressure_payload = await client.read_gatt_char(pressure_char)
+        finally:
+            await client.disconnect()
+        tb_pressure = PRESSURE.get(
+            pressure_payload[0], f"unknown pressure {pressure_payload[0]}"
+        )
+        self.update_sensor(
+            str(OralBSensor.PRESSURE), None, tb_pressure, None, "Pressure"
+        )
+        self.update_sensor(
+            str(OralBSensor.BATTERY_PERCENT), None, battery_payload[0], None, "Battery"
```

Comment:
You can borrow the class from the Govee-ble code

Reply:
Sorry, which class?

Reply:
```diff
diff --git a/src/oralb_ble/parser.py b/src/oralb_ble/parser.py
index 46cb5f0..f0e5f83 100644
--- a/src/oralb_ble/parser.py
+++ b/src/oralb_ble/parser.py
@@ -16,7 +16,7 @@ from bleak_retry_connector import establish_connection, BleakClientWithServiceCa
 from bluetooth_data_tools import short_address
 from bluetooth_sensor_state_data import BluetoothData
 from home_assistant_bluetooth import BluetoothServiceInfo
-from sensor_state_data import SensorUpdate
+from sensor_state_data import SensorUpdate, Units, SensorDeviceClass
 from sensor_state_data.enum import StrEnum
 
 from .const import CHARACTERISTIC_BATTERY, CHARACTERISTIC_PRESSURE
@@ -315,6 +315,10 @@ class OralBBluetoothDeviceData(BluetoothData):
             str(OralBSensor.PRESSURE), None, tb_pressure, None, "Pressure"
         )
         self.update_sensor(
-            str(OralBSensor.BATTERY_PERCENT), None, battery_payload[0], None, "Battery"
+            str(OralBSensor.BATTERY_PERCENT),
+            Units.PERCENTAGE,
+            battery_payload[0],
+            SensorDeviceClass.BATTERY,
+            "Battery",
         )
         return self._finish_update()
```

Reply:
Since this device doesn't use standard unit for pressure, we can't use that

Reply:
Okay that is kind of what I figured. Any other changes still outstanding you need me to make in this PR?